diff --git a/xmousepasteblock.c b/xmousepasteblock.c
index 3bf1f3e..fb8337d 100644
--- a/xmousepasteblock.c
+++ b/xmousepasteblock.c
@@ -56,11 +56,11 @@ void init_eventmask(void) {
     unsigned char mask[(XI_LASTEVENT + 7)/8];
 
     memset(mask, 0, sizeof(mask));
-    masks[0].deviceid = XIAllDevices;
+    masks[0].deviceid = XIAllMasterDevices;
     masks[0].mask_len = sizeof(mask);
     masks[0].mask = mask;
 
-    XISetMask(mask, XI_ButtonPress);
+    XISetMask(mask, XI_RawButtonPress);
 
     XISelectEvents(display, DefaultRootWindow(display), masks, 1);
     XFlush(display);
